node/bft/src/sync/mod.rs

```diff
@@ -97,13 +103,36 @@ impl<N: Network> Sync<N> {
         self.sync_storage_with_ledger_at_bootup().await
     }
 
+    /// Performs one iteration of the block synchronization.
+    #[inline]
+    pub async fn try_block_sync(&self) {
+        // Prepare the block requests, if any.
+        // In the process, we update the state of `is_block_synced` for the sync module.
+        let (block_requests, sync_peers) = self.block_sync.prepare_block_requests();
```

Comment:
is this operation fast? otherwise it should be executed in the context of a blocking task, in order to not block the async worker thread

Reply:
ah, in this case it's unlikely it's an issue, especially since we've profiled BFT performance with `tokio console`; that being said, we might want to do a new run sometime soon, just to be sure we're not blocking anywhere